swim/join_test.go

```diff
@@ -43,6 +48,32 @@ func (s *JoinSenderTestSuite) SetupTest() {
 	s.node = s.tnode.node
 }
 
+func (s *JoinSenderTestSuite) newJoinSenderForDelay() *joinSender {
+	bootstrapHosts := append(fakeHostPorts(1, 2, 1, 1), fakeHostPorts(1, 1, 2, 3)...)
+	joinSender, err := newJoinSender(s.node, &joinOpts{
+		discoverProvider: &StaticHostList{bootstrapHosts},
+	})
+	s.NoError(err, "joiner created successfully")
+	return joinSender
+}
+
+func (s *JoinSenderTestSuite) setupTestForDelay() (*joinSender, [6]float64) {
+	joinSender := s.newJoinSenderForDelay()
+	joinSender.delayOpts.initial = 100
```

Comment:
instead of changing the internal state of the `delayOpts` can we pass in this object to `newJoinSender` via `s.newJoinSenderForDelay`?